cfgrammar/src/lib/yacc/grammar.rs

```diff
@@ -176,7 +176,11 @@ where
         };
 
         for (k, rule) in &ast.rules {
-            rule_names.push((k.clone(), rule.name.1));
+            let ast::Rule {
```

Comment:
I *think* (but I admit I've never tried!) that this can be collapsed into the `for`? So something like:

```rust
for (k, ast::Rule { name: (_, name_span), .. }) in &ast.rules)
```

?

Reply:
Yes, it can!  I had originally had it that way, but it still formats the pattern the same way on a bunch of lines.
That is why I had pulled it out into a let, but happy to pull it up into the for loop.  If after seeing the result we can always reverse course if it's too bad.

Reply:
Good idea to give it a go and see if it's better/worse!

Reply:
I've pushed that, let me know what you think

Reply:
Hmm, very much "in some ways better, in some ways worse"! Let's keep it: we can always change our minds again if we want.